src/cgame/cg_gameinfo.cpp

```diff
@@ -31,12 +31,195 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 #include "common/FileSystem.h"
 #include "cg_local.h"
 
+//
+// arena info
+//
+static int  cg_numArenas;
+static char *cg_arenaInfos[ MAX_ARENAS ];
+
+/*
+===============
+CG_ParseInfos
+===============
+*/
+static int CG_ParseInfos( const char *buf, int max, char *infos[] )
+{
+	char *token;
+	int  count = 0;
+	char key[ MAX_TOKEN_CHARS ];
+	char info[ MAX_INFO_STRING ];
+
+	auto infoPostfixLen = strlen( "\\num\\" ) + strlen( va( "%d", MAX_ARENAS ) );
+
+	while ( 1 )
+	{
+		token = COM_Parse( &buf );
+
+		if ( !token[ 0 ] )
+		{
+			break;
+		}
+
+		if ( strcmp( token, "{" ) )
+		{
+			Log::Notice( "Missing { in info file\n" );
+			break;
+		}
+
+		if ( count == max )
+		{
+			Log::Notice( "Max infos exceeded\n" );
+			break;
+		}
+
+		info[ 0 ] = '\0';
+
+		while ( 1 )
+		{
+			token = COM_ParseExt( &buf, true );
+
+			if ( !token[ 0 ] )
+			{
+				Log::Notice( "Unexpected end of info file\n" );
+				break;
+			}
+
+			if ( !strcmp( token, "}" ) )
+			{
+				break;
+			}
+
+			Q_strncpyz( key, token, sizeof( key ) );
+
+			token = COM_ParseExt( &buf, false );
+
+			if ( !token[ 0 ] )
+			{
+				strcpy( token, "<NULL>" );
+			}
+
+			Info_SetValueForKey( info, key, token, false );
+		}
+
+		//NOTE: extra space for arena number
+		infos[ count ] = (char*) BG_Alloc( strlen( info ) + infoPostfixLen + 1 );
+
+		if ( infos[ count ] )
+		{
+			strcpy( infos[ count ], info );
+			count++;
+		}
+	}
+
+	return count;
+}
+
+/*
+===============
+CG_LoadArenasFromFile
+===============
+*/
+static int CG_LoadArenasFromFile( const char *filename )
+{
+	std::error_code err;
+	std::string text = FS::PakPath::ReadFile( filename, err );
+	if ( err )
+	{
+		Log::Warn( "file not found: %s", filename );
+		return 0;
+	}
+
+	if ( text.size() >= MAX_ARENAS_TEXT )
+	{
+		Log::Warn( "file too large: %s is %i, max allowed is %i",
```

Comment:
This is a pointless error unless you are trying to strictly enforce arena files to have backwards compatibility with some old software.

Reply:
Right, I only want to read legacy files targeting legacy engine, people are expected to write non legacy files. I removed the test and the error.

This allowed me to turn the function as a boolean one.